libminifi/test/unit/ThreadPoolTests.cpp

```diff
@@ -70,8 +70,7 @@ TEST_CASE("ThreadPoolTest1", "[TPT1]") {
   std::function<bool()> f_ex = function;
   utils::Worker<bool> functor(f_ex, "id");
   pool.start();
-  std::future<bool> fut;
-  REQUIRE(true == pool.execute(std::move(functor), fut));
+  auto fut = pool.execute(std::move(functor));
   fut.wait();
```

Comment:
should we assert the validity of the returned future in this test?

Reply:
I'm not sure about whether it makes sense to test for validity or not to be honest. Judging by the documentation https://en.cppreference.com/w/cpp/thread/future/valid `std::future::valid` only returns `false` if `std::future::get` has been called. I can revert the assertion but I assume it would be always `true`.

Reply:
a default constructed future and a future that has been moved from are also `valid() == false`, my only problem is that operating on an "invalid" future is undefined behavior (although encouraged to throw), but I believe we can safely assume that `ThreadPool<T>::execute` returns a valid future, in which case I would remove the validity checks from `SchedulingAgent.cpp`

Reply:
Sorry for the confusing comment. I meant that in this case `valid` would be only `false` after calling `get`/`share`. If I get this right, `std::promise::get_future` can't return an invalid `std::future` since it throws an exception when called on an invalid `std::promise`. E.g. it has no state or `get_future` has already been called once.
I'm not super familiar with `std::promise` to be honest but I think it's safe to remove validity checks in these cases. Wouldn't an invalid `std::future` would have cause problems by now?